include/xtypes/DynamicData.hpp

```diff
@@ -447,6 +615,131 @@ class WritableDynamicDataRef : public ReadableDynamicDataRef
         catch(bool value) { return value; }
     }
 
+    /// \brief Class used for iterate WritableDynamicDataRef
+    class Iterator : public ReadableDynamicDataRef::Iterator
+    {
+    public:
+        Iterator(
+                WritableDynamicDataRef& ref)
+            : ReadableDynamicDataRef::Iterator(ref, false)
+        {}
+
+        WritableDynamicDataRef operator * ()
+        {
+            return ReadableDynamicDataRef::Iterator::operator*();
+        }
+
+    protected:
+
+        friend class WritableDynamicDataRef;
+
+        Iterator(
+                WritableDynamicDataRef& ref,
+                bool end)
+            : ReadableDynamicDataRef::Iterator(ref, end)
+        {
+        }
+    };
+
+    /// \brief Returns the initial iterator of a collection dynamic data.
+    /// \pre The DynamicData must represent a CollectionType.
+    /// \returns The initial iterator.
+    Iterator begin()
+    {
+        assert(type_.is_collection_type());
```

Comment:
Call the `ReadableDynamicDataRef::Iterator` version that already checks this? (The WritableOperator could be getted by calling implicitly a protected constructor that creates writable iterator from a readable iterator).